algebird-core/src/main/scala/com/twitter/algebird/CorrelationGroup.scala

```diff
@@ -0,0 +1,107 @@
+ *
+ */
+case class Correlation(
+    c2: Double,
+    m2Left: Double,
+    m2Right: Double,
+    m1Left: Double,
+    m1Right: Double,
+    m0: Long
+) {
+  def count: Long = m0
+
+  def meanLeft: Double = m1Left
+
+  def meanRight: Double = m1Right
+
+  // useful in calculating variance and stddev; private because we don't have skewness or kurtosis info
+  private[algebird] def leftMoment: Moments = Moments(m0 = m0, m1 = m1Left, m2 = m2Left, 0, 0)
+
+  private[algebird] def rightMoment: Moments = Moments(m0 = m0, m1 = m1Right, m2 = m2Right, 0, 0)
+
+  // variance, stddev, and covariance are for the population, not a sample
+  def varianceLeft: Double = leftMoment.variance
+
+  def varianceRight: Double = rightMoment.variance
+
+  def stddevLeft: Double = leftMoment.stddev
+
+  def stddevRight: Double = rightMoment.stddev
+
+  def covariance: Double = c2 / count
+
+  /**
+   *
+   * @return Pearson's correlation coefficient
+   */
+  def correlation: Double =
+    // correlation is defined as: covariance / (varianceLeft * varianceRight)
+    // however, dividing by "count" cancels out, and leaves us with the following formula, which relies on fewer
+    // divisions
+    c2 / (Math.sqrt(m2Left * m2Right))
+}
+
+object CorrelationGroup extends Group[Correlation] {
+
+  /**
+   * The algorithm for combining the correlation calculations from two partitions of pairs of numbers. Comes from
+   * Pébay, Philippe (2008), "Formulas for Robust, One-Pass Parallel Computation of Covariances and Arbitrary-Order Statistical Moments",
+   *   Technical Report SAND2008-6212, Sandia National Laboratories
+   * https://prod-ng.sandia.gov/techlib-noauth/access-control.cgi/2008/086212.pdf
+   */
+  override def plus(a: Correlation, b: Correlation): Correlation = {
+    val count = a.count + b.count
+    if (count == 0)
+      zero
+    else {
+      val prodSumRatio = a.count.toDouble * b.count / count
+      val m1Left = MomentsGroup.getCombinedMean(a.count, a.m1Left, b.count, b.m1Left)
+      val m1Right = MomentsGroup.getCombinedMean(a.count, a.m1Right, b.count, b.m1Right)
+      val deltaLeft = b.m1Left - a.m1Left
+      val deltaRight = b.m1Right - a.m1Right
+
+      val m2Left = a.m2Left + b.m2Left + math.pow(deltaLeft, 2) * prodSumRatio
+      val m2Right =
+        a.m2Right + b.m2Right + math.pow(deltaRight, 2) * prodSumRatio
+
+      val c2 = a.c2 + b.c2 + deltaLeft * deltaRight * prodSumRatio
+
+      Correlation(c2 = c2, m2Left = m2Left, m2Right = m2Right, m1Left = m1Left, m1Right = m1Right, m0 = count)
+    }
+  }
+
+  override val zero = Correlation(0, 0, 0, 0, 0, 0)
+
+  override def negate(v: Correlation): Correlation =
+    Correlation(
+      c2 = -v.c2,
+      m2Left = -v.m2Left,
+      m2Right = -v.m2Right,
+      m1Left = v.m1Left,
+      m1Right = v.m1Right,
+      m0 = -v.m0
+    )
+
+}
+
+object CorrelationAggregator extends MonoidAggregator[(Double, Double), Correlation, Correlation] {
```

Comment:
seems like you might want the `.andThenPresent(_.correlation)` a lot.

Reply:
Agreed, but sometimes you might want `.andThenPresent(_.covariance)` from time-to-time as well. Maybe I can add a convenience method to this object?

Reply:
good point. Yeah, two convenience methods on this would be good. `def correlation` and `def covariance` would be good.